fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -1208,11 +1208,11 @@ void testSimpleUpdate(boolean primaryKeySegmentIndexEnabled) {
                         .getDirectory(Tuple.from())
                         .getPrimaryKeySegmentIndex();
                 assertEquals(List.of(
```

Comment:
We should add tests that:
1. if we delete all the documents the FieldInfos space is empty
2. after a merge that the reference counting for FieldInfos is correct

Reply:
I added the tests / assertions, and it found a bug with the delete code, where it was deleting the wrong id. Since we use the same counter to calculate ids, this wouldn't every break anything, but also means that we never deleted FieldInfos data.